Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -0,0 +1,88 @@
+import UIKit
+
+// MARK: CameraAnimator Class
+public class CameraAnimator: NSObject {
+
+    // MARK: Stored Properties
+
+    /// Instance of the property animator that will run animations.
+    private let propertyAnimator: UIViewPropertyAnimator
+
+    /// Delegate that conforms to `CameraAnimatorDelegate`.
+    private var delegate: CameraAnimatorDelegate?
+
+    /// The ID of the owner of this `CameraAnimator`.
+    internal var owner: AnimationOwnerProtocol
```

Comment:
should this be weak?

Reply:
Its an enum so I don't think so

Reply:
we're passing in an enum but what happens if a developer conforms to it themselves and passes in an object?

Reply:
Yes that would possibly cause a retain cycle, if not handled correctly.

Reply:
One complication here is that in order for that to be weak, then the protocol would need to be `:class` so then `AnimationOwner` can't be an enum

Reply:
Hmm, maybe we get rid of the protocol then and just use the enum directly.

Reply:
I think that makes a lot of sense, no reason to expose that protocol right now.

Reply:
sounds good, let's remove the protocol for now.